lib/appsignal/integrations/delayed_job_plugin.rb

```diff
@@ -54,6 +53,20 @@ def self.invoke_with_instrumentation(job, block)
         end
       end
 
+      def self.class_and_method_name_from_object_or_hash(payload, default_name)
+        # Attempt to find appsignal_name override
+        class_and_method_name = extract_value(payload, :appsignal_name, nil)
+        return class_and_method_name.split("#") if class_and_method_name.is_a?(String)
+
+        pound_split = default_name.split("#")
+        return pound_split if pound_split.length == 2
+
+        dot_split = default_name.split(".")
+        return dot_split if dot_split.length == 2
```

Comment:
This will break the original behavior. `Class.foo` will be reported as `Class#foo` because of the behavior in https://github.com/appsignal/appsignal-ruby/blob/0899c80a741678209a06e9d76094a2b079610372/lib/appsignal/transaction.rb#L221

Robert will make a patch to restore the original behavior after #612 is merged.

Reply:
Fixed in #613